jenkins/ci.suse.de/openstack-ardana-pcloud.yaml

```diff
@@ -74,74 +77,89 @@
 
-      - string:
+      - validating-string:
           name: controllers
           default: '3'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of controller nodes in the generated input model.
+            The number of controller nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: standard, entry-scale-kvm.
 
-      - string:
+      - validating-string:
           name: core_nodes
           default: '2'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of OpenStack core services nodes in the generated input model.
+            The number of OpenStack core services nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: mid-scale-kvm, std-split.
 
-      - string:
+      - validating-string:
           name: lmm_nodes
           default: '3'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of LMM services nodes in the generated input model.
+            The number of LMM services nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: mid-scale-kvm, std-split.
 
-      - string:
+      - validating-string:
           name: dbmq_nodes
           default: '3'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of database & rabbitmq service nodes in the generated input model.
+            The number of database & rabbitmq service nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: mid-scale-kvm, std-split.
 
-      - string:
+      - validating-string:
           name: neutron_nodes
           default: '2'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of neutron network nodes in the generated input model.
+            The number of neutron network nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: mid-scale-kvm.
 
-      - string:
+      - validating-string:
           name: swift_nodes
           default: '3'
+          regex: '[0-3]'
+          msg: The entered value failed validation
           description: |
-            The number of swift proxy/account/container/object service nodes in the generated input model.
+            The number of swift proxy/account/container/object service nodes in the generated input model (0-3).
 
             Input model generator scenarios using this parameter: mid-scale-kvm.
 
-      - string:
+      - validating-string:
           name: sles_computes
           default: '2'
+          regex: '[0-5]'
```

Comment:
For the scale environment we have 200+ nodes

Reply:
Right. Good point.